### ISSUE TITLE
VersionAndroid not working with different naming groups

### ISSUE THREAD

**Issue by ChrisLGardner** [rfennell/vNextBuild#222]
Raised via the Marketplace:

I have a question about VersionAndroidManifest
It works fine, when I set the version number format to {1}.{2} and version code format to {3}

Filename Pattern: AndroidManifest.xml
Version Number/Build Number: 2017.12.5.2_11762
Version Filter to extract build number: \d+\.\d+\.\d+\.\d+\_\d+
Version Number Format: {1}.{2}
Version Code Format: {3}
Output: Version Number Parameter Name: OutputedVersion
Found 0 files to update.
Extracted Version: 2017.12.5.2_11762
Version Name will be: 2017.12
Version Code will be 5

But it does not work, if I change version number format to {1}.{2}.{3}.{4} and version code format to {5}.
What am I doing wrong? 

Filename Pattern: AndroidManifest.xml
Version Number/Build Number: 2017.12.5.1_11760
Version Filter to extract build number: \d+\.\d+\.\d+\.\d+\_\d+
Version Number Format: {1}.{2}.{3}.{4}
Version Code Format: {5}
Output: Version Number Parameter Name: OutputedVersion
Found 0 files to update.
Extracted Version: 2017.12.5.1_11760
Version Name will be: 2017.12.5.1_11760
Version Code will be

**Comment by ChrisLGardner** [comment]
After some digging it turns out the reason is that we split the version number on '.'. Might need to look at possibly splitting on other characters with a parameter but that's going to take some work to figure out the best way to handle it and what happens when we do (in terms of numbering array items).

**Comment by rfennell** [comment]
Would this do the job?

```
$myString="2017.12.5.1_11760"
$arr = $myString -split {$_ -eq "." -or $_ -eq "_"} 
$arr # Print output
```
It would mean passing in parameters in a different way, a breaking change

**Comment by rfennell** [comment]
OK now in NodeJS

```
   const version = "2017.12.5.1_11760";
   const versionNumberSplit = version.split(/[._]/);
   expect(versionNumberSplit[0]).to.equal("2017");
   expect(versionNumberSplit[1]).to.equal("12");
   expect(versionNumberSplit[2]).to.equal("5");
   expect(versionNumberSplit[3]).to.equal("1");
   expect(versionNumberSplit[4]).to.equal("11760");
```

We will need to edit the VersionNumber and VersionCode functions as these hard code the separator as .

We could move over to this hard coding of . and _, but would it be better to try to detect this from the regex expression passed as parameter?

**Comment by rfennell** [comment]
How does this look?

```
       const format = "\d+.\d+.\d+.\d+_\d+";
        const delimiters = format.replace("\d+", "");

        const version = "2017.12.5.1_11760";
        const versionNumberSplit = version.split(new RegExp("[" + delimiters + "]"));

        expect(versionNumberSplit[0]).to.equal("2017");
        expect(versionNumberSplit[1]).to.equal("12");
        expect(versionNumberSplit[2]).to.equal("5");
        expect(versionNumberSplit[3]).to.equal("1");
        expect(versionNumberSplit[4]).to.equal("11760");

```

**Comment by ChrisLGardner** [comment]
That looks good for generating the delimited based on the input format. That should cover almost all the cases we're likely to see.

**Comment by rfennell** [comment]
OK I will get that out in the next few days

**Comment by rfennell** [comment]
Released as 1.30..2
